src/Random.mo

```diff
@@ -8,64 +18,207 @@ module {
 
   public let blob : shared () -> async Blob = rawRand;
 
-  /// Opinionated choice of PRNG from a given seed.
-  public func new(seed : Blob) : Random {
-    todo()
+  /// Creates a fast pseudo-random number generator using the SFC64 algorithm.
+  /// This provides statistical randomness suitable for simulations and testing,
+  /// but should not be used for cryptographic purposes.
+  /// The seed blob's first 8 bytes are used to initialize the PRNG.
+  public func fast(seed : Nat64) : Random {
+    let prng = PRNG.sfc64a();
+    prng.init(seed);
+    Random(
+      func() {
+        // Generate 8 bytes directly from a single 64-bit number
+        let n = prng.next();
+        // TODO: optimize using Array.tabulate or even better: a new primitive
```

Comment:
I totally agree that this should be a prim. `toNat` followed by `fromNat` is wasteful.